vision_agent_tools/tools/florence2_sam2.py

```diff
@@ -0,0 +1,144 @@
+from dataclasses import dataclass
+from typing_extensions import Annotated
+
+import torch
+import numpy as np
+from PIL import Image
+from pydantic import validate_call
+
+from vision_agent_tools.shared_types import BaseTool, VideoNumpy, SegmentationBitMask
+from vision_agent_tools.tools.florencev2 import Florencev2, PromptTask
+
+from sam2.sam2_video_predictor import SAM2VideoPredictor
+from sam2.sam2_image_predictor import SAM2ImagePredictor
+
+
+_HF_MODEL = "facebook/sam2-hiera-large"
+
+
+@dataclass
+class ImageBboxAndMaskLabel:
+    label: str
+    bounding_box: list[
+        Annotated[float, "x_min"],
+        Annotated[float, "y_min"],
+        Annotated[float, "x_max"],
+        Annotated[float, "y_max"],
+    ]
+    mask: SegmentationBitMask | None
+
+
+@dataclass
+class MaskLabel:
+    label: str
+    mask: SegmentationBitMask
+
+
+class Florence2SAM2(BaseTool):
+    def __init__(self):
+        self.florence2 = Florencev2()
+        self.video_predictor = SAM2VideoPredictor.from_pretrained(_HF_MODEL)
+        self.image_predictor = SAM2ImagePredictor(self.video_predictor)
+
+    @torch.inference_mode()
+    def get_bbox_and_mask(
+        self, image: Image.Image, prompts: list[str], return_mask: bool = True
+    ) -> dict[int, ImageBboxAndMaskLabel]:
+        objs = {}
+        self.image_predictor.set_image(np.array(image, dtype=np.uint8))
+        annotation_id = 0
+        for prompt in prompts:
+            with torch.autocast(device_type="cuda", dtype=torch.float16):
```

Comment:
@dillonalaird I think this would be better to define a `self.device` value that stores the device type instead of hard coding it.

Reply:
thanks! done